applications/serializers.py

```diff
@@ -105,10 +106,19 @@ class BootcampApplicationSerializer(serializers.ModelSerializer):
     """BootcampApplication serializer"""
 
     bootcamp_run = BootcampRunSerializer()
+    has_payments = serializers.SerializerMethodField()
+
+    def get_has_payments(self, application):
+        """Return true if the user has any fulfilled orders for this application"""
+        # IMPORTANT: orders has already been prefetched and filtered status=Order.FULFILLED
```

Comment:
Can we just pass in a context variable to the serializer that indicates that the application is prefetched? Then in the body of this method we can have a conditional around the value of that context variable. If true, use this `bool()` statement. If false, run the actual query and filter on FULFILLED. It will be useful to have a comment to explain it, but you'll probably want to edit the comments you have here now